### PR TITLE
core: forbid duplicate ring members from v6

### PR DESCRIPTION
**Pull request by moneromooo-monero**
This avoids someone adding what amounts to mixin 0 rings in
practice, as there is no other good reason to allow this.

### CODE REVIEW

File: src/cryptonote_core/cryptonote_core.cpp [hunk 1]
```diff
@@ -752,6 +758,22 @@ namespace cryptonote
...
+  bool core::check_tx_inputs_ring_members_diff(const transaction& tx) const
+  {
+    const uint8_t version = m_blockchain_storage.get_current_hard_fork_version();
+    if (version >= 6)
+    {
+      for(const auto& in: tx.vin)
+      {
+        CHECKED_GET_SPECIFIC_VARIANT(in, const txin_to_key, tokey_in, false);
+        for (size_t n = 1; n < tokey_in.key_offsets.size(); ++n)
+          if (tokey_in.key_offsets[n] == 0)
```
Comment (vtnerd) [inline src/cryptonote_core/cryptonote_core.cpp]: I think this check is insufficient:
- The key_offsets can possibly be randomized based on another patch.
- This does not catch the case where neither offset is the first one.

I think the easiest method is to copy the `std::vector` -> `std::sort(copy.begin(), copy.end())` -> `if (std::unique(copy.begin(), copy.end()) != copy.end()) { return false }`.

Comment (moneromooo-monero) [inline src/cryptonote_core/cryptonote_core.cpp]: I think it is sufficient. key_offsets is unsigned. These are relative offsets. I don't understand which two offsets you are talking about in the second point. I don't understand the point of the copy.assign comment, since this doesn't do copies.

Comment (vtnerd) [inline src/cryptonote_core/cryptonote_core.cpp]: Another bad comment from review day. The shuffling patch mixed me up a bit / forgot these are relative offsets. It would be nice if the C++ type system were used a bit better to indicate format (this is a complaint of the existing codebase in general).
